Dockerfile

```diff
@@ -0,0 +1,54 @@
+# Use Python 3.12 slim image
+FROM python:3.12-slim-bookworm
+
+# Set environment variables
+ENV PYTHONUNBUFFERED=1 \
+    PYTHONDONTWRITEBYTECODE=1 \
+    PIP_NO_CACHE_DIR=1 \
+    PIP_DISABLE_PIP_VERSION_CHECK=1
+
+# Create app user and group
+RUN groupadd --gid 1000 app && \
+    useradd --uid 1000 --gid app --shell /bin/bash --create-home app
+
+# Install system dependencies
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        curl \
+        git \
+        build-essential \
+        && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Set working directory
+WORKDIR /app
+
+# Copy requirements and install Python dependencies
+COPY requirements-dev.txt .
+RUN pip install --no-cache-dir -r requirements-dev.txt
+
+# Copy source code
+COPY src/ ./src/
+COPY pyproject.toml .
+COPY README.md .
+COPY LICENSE .
+
+# Install the package in development mode
+RUN pip install -e .
+
+# Change ownership to app user
+RUN chown -R app:app /app
+
+# Switch to app user
+USER app
+
+# Expose ports
+EXPOSE 2218
+
+# Health check
+HEALTHCHECK --interval=30s --timeout=5s --start-period=30s --retries=3 \
+    CMD curl -f http://localhost:2218/health || exit 1
+
+# Default command
+CMD ["launchpad", "devserver"]
```

Comment:
Just to confirm - is this a dockerfile that is used to build production images? If so it should likely not be using a devserver but a more production ready server

Reply:
I was under the assumption that for production we ideally use the production image from the docker file, but frankly I'm not sure if that's the case. WDYT?

> If so it should likely not be using a devserver but a more production ready server
heh, i just copied snuba's "devserver" naming, despite it originally being "server". I don't think I have any difference in config between dev and prod yet

Reply:
synced offline, keeping the dockerfile around precisely for the prod use case